### PR TITLE
Keep exemptions in case of closing a tab and subsequently undoing the action

### PR DESCRIPTION
**Pull request by anikiki**
Task/Issue URL: https://app.asana.com/0/488551667048375/1207991167374138/f

### Description
Updated the code to make sure the exemption is kept after undoing a tab close action.

### Steps to test this PR
See the **Tests** section in the description of https://app.asana.com/0/488551667048375/1207991167374138/f

### CODE REVIEW

**Comment by anikiki** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4905?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4905** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4905?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4903** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4903?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4899** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4899?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#4898** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/4898?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @anikiki and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: app/src/main/java/com/duckduckgo/app/tabs/ui/TabSwitcherViewModel.kt [hunk 1]
```diff
@@ -126,6 +126,10 @@ class TabSwitcherViewModel @Inject constructor(
     }
 
     suspend fun purgeDeletableTabs() {
+        Timber.d("TAG_ANA Purging deletable tabs")
```
Comment (marcosholgado) [inline app/src/main/java/com/duckduckgo/app/tabs/ui/TabSwitcherViewModel.kt:129]: 👀

**Review by marcosholgado — APPROVED** [review]
LGTM

File: app/src/main/java/com/duckduckgo/app/tabs/ui/TabSwitcherViewModel.kt (hunk 1 above)
Comment (anikiki) [reply app/src/main/java/com/duckduckgo/app/tabs/ui/TabSwitcherViewModel.kt:129]: It's removed in another PR :see_no_evil:

**Pull request merged**